dotcom-rendering/src/web/experiments/tests/sign-in-gate-mandatory-long-testrun.ts

```diff
@@ -14,25 +14,26 @@ export const signInGateMandatoryLongTestRunUk: ABTest = {
 	description:
 		'Test run for long mandatory test - Show sign in gate to global users on 3rd article view of simple article templates, with higher priority over banners and epic.',
 	audience: 0.0025,
-	audienceOffset: 0.89,
+	audienceOffset: 0.89, // 890001 - 892500
 	successMeasure: 'Users sign in or create a Guardian account',
 	audienceCriteria:
 		'Global, 3rd article of the day, lower priority than consent banner, simple articles (not gallery, live etc.), not signed in, not shown after dismiss, not on help, info sections etc. Exclude iOS 9 and guardian-live-australia. Suppresses other banners, and appears over epics',
 	dataLinkNames: 'SignInGateMandatoryLongTestRunUk',
 	idealOutcome:
 		'Increase the number of users signed in whilst running at a reasonable scale',
 	showForSensitive: false,
-	canRun: () =>
-		['GB'].includes(getCountryCodeSync() ?? '') &&
-		setOrUseParticipations(
-			setParticipationsFlag,
-			'SignInGateMandatoryLongTestRunUk', // test id
-			'mandatory-long-testrun-uk', // variant id - can only be used for single variant test
-		),
+	canRun: () => true,
 	variants: [
 		{
 			id: 'mandatory-long-testrun-uk',
 			test: (): void => {},
+			canRun: () =>
+				['GB'].includes(getCountryCodeSync() ?? '') &&
+				setOrUseParticipations(
+					setParticipationsFlag,
+					'SignInGateMandatoryLongTestRunUk', // test id
+					'mandatory-long-testrun-uk', // variant id - can only be used for single variant test
+				),
```

Comment:
Had to move this to the Variant's optional `canRun` function because of the [order in which the ab testing framework calcutes what it can run](https://github.com/guardian/ab-testing/blob/9cfc419c49f92d08bc41c9185a1642602d0e3164/packages/ab-core/src/core.ts#L110-L115) - which is:

1/  Forced test exceptions
2/ ABTest definition `canRun` function
3/ Test Variant based on `mvt_id` cookie
4/ Variant definition `canRun` function if present

If we used the test's canRun function we risk setting the participation for a test that falls outside of the present `mvt_id` cookie bucket range but is in the correct reason.

This is only the case on non overlapping tests.

Reply:
I think the canRun method of a test is optional, so you could remove it, if unused.

Reply:
Its only optional on the Variant type, required on the ABTest type so this needs to default to true